ch_backup/ch_backup.py

```diff
@@ -31,10 +31,39 @@
 from ch_backup.logic.database import DatabaseBackup
 from ch_backup.logic.table import TableBackup
 from ch_backup.logic.udf import UDFBackup
-from ch_backup.util import now, utcnow
+from ch_backup.util import cached_property, now, utcnow
 from ch_backup.version import get_version
 
 
+class ContextLazyLoader:
+    """
+    Lazy loader for BackupContext
+    """
+
+    def __init__(self, config: Config) -> None:
+        self._config = config
+
+    @cached_property
+    def context(self) -> BackupContext:
```

Comment:
Why not just move this method in `ClickhouseBackup` class. And in `reload_config()` just clear its cache
`self.context.fget.cache_clear()`.

`ContextLazyLoader` class seems too specialized.

Reply:
Fixed